synapse/storage/databases/state/store.py

```diff
@@ -228,6 +247,144 @@ def _get_state_for_group_using_cache(
 
         return state_filter.filter_state(state_dict_ids), not missing_types
 
+    def _get_state_for_group_gather_inflight_requests(
+        self, group: int, state_filter: StateFilter
+    ) -> Tuple[Sequence[ObservableDeferred[StateMap[str]]], StateFilter]:
+        """
+        Attempts to gather in-flight requests and re-use them to retrieve state
+        for the given state group, filtered with the given state filter.
+
+        Returns:
+            tuple (
+                sequence of ObservableDeferreds to observe,
+                StateFilter representing what is left over (what else needs
+                to be requested to fulfil the request)
+            )
```

Comment:
I tried to somehow make this so it looks like two items, not three:

```suggestion
            Tuple of two values:
                A sequence of ObservableDeferreds to observe
                A StateFilter representing what else needs to be requested to fulfill the request
```